.github/workflows/pull_request_tests.yaml

```diff
@@ -0,0 +1,40 @@
+name: Test Action in CI
+
+on:
+  pull_request:
+    branches:
+      - master
+
+env:
+  CLANG_TIDY_VERSION: 17.0.1
+  CMAKE_VERSION: 3.21.3
+  # cannot use > 3.11 due to https://github.com/ssciwr/clang-tidy-wheel/issues/66
+  PYTHON_VERSION: 3.11
+
+jobs:
+  check_action:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-python@v5
+        with:
+          python-version: "${{ env.PYTHON_VERSION }}"
+      - name: Install dependencies
+        run: |
+          pip install clang-tidy==${CLANG_TIDY_VERSION} cmake==${CMAKE_VERSION}
```

Comment:
Interesting choice that you went with `pip` to install these dependencies. Any particular benefit compared to `apt`?

Reply:
1. Cross platform
2. The baseline image for Ubuntu always lags behind in clang and cmake versions. 
3. Using the clang-tidy wheel from https://github.com/ssciwr/clang-tidy-wheel future proofs this test in case we'd ever want to run multiple versions in a matrix.